### ISSUE TITLE
Add Trample as an Attack type

### ISSUE THREAD

**Issue by sIKE23** [sIKE23/Mage-Wars#346]
Needed for Lost Grimore, the [Trample] is the name of the attack in the AttackBar of the set data.

**Comment by ACG8** [comment]
Aside from storing it in a dictionary (as a rangetype), is anything else needed here?

**Comment by sIKE23** [comment]
Just want to be able to draw an Arrow and use the attack....

**Comment by sIKE23** [comment]
### Trample
Once per round, when a creature moves out of a zone, it may make a trample attack as a free action against a target creature in its zone. This means that the attack can only be used when you take a move action to leave the zone. The attack occurs during the move action, during the “leaving zone effects step”. This means you may only attack a creature in the zone you are leaving, not the zone you are entering. There is a special trample icon for trample attacks.

Trample attacks are considered melee attacks and must attack eligible targets. However, trample attacks are not affected by traits that would normally affect melee attacks. This includes Melee +X, Melee -X, Bloodthirsty +X, Charge +X, or any other traits that may affect melee attacks. After moving, your creature is still able to take a quick action as normal, which might be making another melee attack.
Trample attacks may only

**Comment by sIKE23** [comment]
@ACG8 
The important part is 
```
 trample attacks are not affected by traits that would normally affect melee attacks. This includes Melee +X, Melee -X, Bloodthirsty +X, Charge +X, or any other traits that may affect melee attacks.
```

**Comment by ACG8** [comment]
Ah. It is inconvenient that they are considered melee attacks. I guess I will make trample an action type instead of a rangetype.

**Comment by sIKE23** [comment]
They are much like counterstrike from a implementation perspective....

**Comment by ACG8** [comment]
I can't sync for some reason, and am not sure how to debug the repo. Can you think of anything that might be the problem?

**Comment by sIKE23** [comment]
@ACG8 
Bear Strength is buffing the Titanodons Trample attack still....

**Comment by ACG8** [comment]
I found the issue. It is a problem with the formatting in the xml file. The relevant line currently reads

```<property name="AttackBar" value="Gore:&#xD;&#xA;[Quick] [Melee] [Dice=5] [+2 vs. Corporeal Conjurations]&#xD;&#xA;Thundering Footsteps:&#xD;&#xA; [Trample] [Melee] [Dice=7]"```

But it should read
```<property name="AttackBar" value="Gore:&#xD;&#xA;[Quick] [Melee] [Dice=5] [+2 vs. Corporeal Conjurations]&#xD;&#xA;Thundering Footsteps:&#xD;&#xA;[Trample] [Melee] [Dice=7]"```

In other words, there should not be a space before ```[Trample]```, as this is causing the character-stripping step (in which ```[``` and ```]``` characters on the outside of a string are removed) to fail.

Since this xml file is not currently included in the branch (due to NDA reasons, I assume), I will not upload a commit but will instead rely on you fix the problem (the issue is with the file I got from the playtester dropbox, and I can't edit that.)

Basically, there should be no spaces between the name of the attack and the first trait of the attack.
